core-bundle/contao/widgets/SerpPreview.php

```diff
@@ -41,7 +41,7 @@ public function generate()
 		}
 
 		$id = $model->id;
-		$title = StringUtil::substr($this->getTitle($model), 64);
+		$title = StringUtil::substr(str_replace(array('[nbsp]', '[-]'), array(' ', ''), $this->getTitle($model)), 64);
```

Comment:
This should not be needed as we do not store basic entities in the database anymore in Contao 5, see [#4453](https://github.com/contao/contao/pull/4453#issuecomment-1094112590)

Reply:
If `['eval' => ['basicEntities' => true]]` is enabled for that field, you can still enter `[nbsp]` and `[-]`. But the value that is stored in the database (and in `$model` in this case) is the actual entity, so `&nbsp;` and `&shy;`.

Reply:
But this also could lead to strings like `…Dampf&sh`. Therefore I would suggest striping any `&shy;` and `&nbsp;` from `$this->getTitle($model)` before shortening it:

```php
$title = StringUtil::substr(str_replace(array('&nbsp;', '&shy;'), array(' ', ''), $this->getTitle($model)), 64);
```

In mootao.js this would make be `title.replace(/\&shy;/g, '').replace(/\&nbsp;/g, ' ')` instead of `title.replace(/\[-\]/g, '').replace(/\[nbsp\]/g, ' ')`

Reply:
> In mootao.js this would make be title.replace(/\&shy;/g, '').replace(/\&nbsp;/g, ' ') instead of title.replace(/\[-\]/g, '').replace(/\[nbsp\]/g, ' ')

Wrong: in mootao.js `title.replace(/\[-\]/g, '').replace(/\[nbsp\]/g, ' ')` is the right way since the _Basic Entities_ are still in use…

Reply:
Actually [`StringUtil::substr()`](https://github.com/contao/contao/blob/5.x/core-bundle/contao/library/Contao/StringUtil.php#L29) does handle entities:

> The function preserves words, so the result might be a bit shorter or longer than the number of characters given.

So `Dampf&shy;schiff` or `1000&nbsp;€` would be preserved. I am just not sure how Google shortens the title on the result page. Therefore I think we can just replace `&nbsp;` with ` ` [space] and remove `&shy;`.

Reply:
I think we should just keep all entities here, nothing special about `nbsp` and `shy` IMO.